serving/docker/partition/sm_neo_quantize.py

```diff
@@ -0,0 +1,115 @@
+
+import sys
+import logging
+import os
+from types import SimpleNamespace
+from typing import Final
+import torch
+
+from sm_neo_utils import (CompilationFatalError, write_error_to_file,
+                          get_neo_env_vars)
+from utils import extract_python_jar
+#from quantization_properties_manager import QuantizationPropertiesManager
+#from quantize import QuantizationService
+from properties_manager import PropertiesManager
+from partition import PartitionService
+
+PYTHON_CACHE_DIR = '/tmp/djlserving/cache'
+
+
+class NeoQuantizationService():
+
+    def __init__(self):
+        self.args: SimpleNamespace = SimpleNamespace()
+        self.properties_manager: PropertiesManager = None
+        self.compiler_flags: dict = None
+
+        env = get_neo_env_vars()
+        self.INPUT_MODEL_DIRECTORY: Final[str] = env[1]
+        self.OUTPUT_MODEL_DIRECTORY: Final[str] = env[2]
+        self.COMPILATION_ERROR_FILE: Final[str] = env[3]
+        self.COMPILER_CACHE_LOCATION: Final[str] = env[4]
+
+    def update_dataset_cache_location(self):
+        logging.info(
+            f"Updating HuggingFace Datasets cache directory to: {self.COMPILER_CACHE_LOCATION}"
+        )
+        os.environ['HF_DATASETS_CACHE'] = self.COMPILER_CACHE_LOCATION
+        #os.environ['HF_DATASETS_OFFLINE'] = "1"
+
+    def initialize_partition_args_namespace(self):
+        """
+        Initialize args, a SimpleNamespace object that is used to instantiate a
+        PropertiesManager for partitioning. PropertiesManager expects an
+        argparse.Namespace, but we use a SimpleNamespace in its place because it
+        is easier to construct.
+
+        These attributes are defined in the partition.py argparser.
+        PropertiesManager expects these attributes to be defined to be initialized.
+        """
+        self.args.save_mp_checkpoint_path = self.OUTPUT_MODEL_DIRECTORY
+        num_gpus = torch.cuda.device_count()
+        self.args.tensor_parallel_degree = num_gpus
+        self.args.properties_dir = self.INPUT_MODEL_DIRECTORY
+        self.args.model_id = None
+        self.args.quantize = None
+
+    def construct_properties_manager(self):
+        """
+        Factory method used to construct a QuantizationPropertiesManager from
+        given serving.properties
+        """
+        # Default to awq quantization
+        os.environ['OPTION_QUANTIZE'] = 'awq'
+        logging.debug("Constructing PropertiesManager from "
+                      f"serving.properties\nargs:{self.args}\n")
+        self.properties_manager = PropertiesManager(self.args)
+
+    def run_quantization(self) -> str:
+        """
+        :return: the output of the partition command captured from stdout
+        """
+        partition_service = PartitionService(self.properties_manager)
+        extract_python_jar(PYTHON_CACHE_DIR)
+        try:
+            return partition_service.run_partition()
```

Comment:
Isnt  `partition_service.run_quantization()` here?

Reply:
Yes, thank you.